hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -472,18 +472,22 @@ onOpenNetworkReqSn env ledger st otherParty sv sn requestedTxIds mDecommitTx =
       unseen -> wait $ WaitOnTxs unseen
 
   requireApplicableDecommitTx cont =
-    case mDecommitTx of
-      Nothing -> cont (confirmedUTxO, Nothing)
-      Just decommitTx ->
-        -- Spec: require ̅S.𝑈 ◦ txω /= ⊥
-        case applyTransactions ledger currentSlot confirmedUTxO [decommitTx] of
-          Left (_, err) ->
-            Error $ RequireFailed $ SnapshotDoesNotApply sn (txId decommitTx) err
-          Right newConfirmedUTxO -> do
-            -- Spec: 𝑈_active ← ̅S.𝑈 ◦ txω \ outputs(txω)
-            let utxoToDecommit = utxoFromTx decommitTx
-            let activeUTxO = newConfirmedUTxO `withoutUTxO` utxoToDecommit
-            cont (activeUTxO, Just utxoToDecommit)
+    -- NOTE: if decommit tx was already applied when we constructed the previous
+    -- snapshot we should not try to apply it again
+    if confUTxOToDecommit == decommitUTxO
```

Comment:
Although possible.. why do we re-request the `decommitTx` in this snapshot if it was already applied?

IMO it would be more intuitive to not even request it?

In any case, we **must** update the spec to match this. It's something we have overlooked in the algorithm.

Reply:
I was thinking the same thing: Is it ok to just not include the decommit tx in the next snapshot if we already applied it? One question is how do we know if we applied it or not? we would have to keep some information about it in the local state I think. I would prefer if we don't populate the `utxoToDecommit` in the snapshot if it is not related to the snapshot in question. 

I think I need to draw this on the miro board to see what kind of scenarios are possible when you keep the decommit utxo around vs. when you omit it.